src/renderer/dispatchers/app.ts

```diff
@@ -17,12 +17,66 @@ import {
   setManagedBank,
   setManagedValidator,
 } from '@renderer/store/app';
-import {AddressData, AppDispatch, RootState, ValidatorConfig} from '@renderer/types';
+import {AddressData, AppDispatch, BankConfig, ManagedNode, Protocol, RootState, ValidatorConfig} from '@renderer/types';
 import {formatAddressFromNode} from '@renderer/utils/address';
 
 import {fetchBankConfig} from './banks';
 import {fetchValidatorConfig} from './validators';
 
+interface ManagedBankProps {
+  bankConfig: BankConfig;
+  managedBank: ManagedNode;
+}
+
+const getManagedBankAddress = ({bankConfig, managedBank}: ManagedBankProps): AddressData => {
+  if (bankConfig.protocol.includes(Protocol.HTTPS)) {
+    return bankConfig;
+  }
+  if (managedBank.protocol.includes(Protocol.HTTPS)) {
+    return {
+      ip_address: managedBank.ip_address,
+      port: undefined,
+      protocol: managedBank.protocol,
+    };
+  }
+
+  return bankConfig;
+};
+
+interface DefaultResponse {
+  address: string;
+  error: any;
+  bankConfig?: undefined;
```

Comment:
same for `bankConfig` and `validatorConfig`, are the types needed?

Reply:
https://github.com/thenewboston-developers/Account-Manager/pull/634#discussion_r719161499